src/main/java/software/amazon/awssdk/crt/mqtt5/Mqtt5Client.java

```diff
@@ -202,6 +227,48 @@ private synchronized void setIsConnected(boolean connected) {
         isConnected = connected;
     }
 
+
+    /*******************************************************************************
+     * Mqtt5 to Mqtt3 Adapter
+     ******************************************************************************/
+
+    /**
+     * Returns a MqttConnection Object based on Mqtt5Client configuration. A client id is required for NewConnection.
+     * @param callbacks connection event callbacks to use
+     *
+     * @return a MqttClientConnection
+     */
+    public MqttClientConnection NewConnection(MqttClientConnectionEvents callbacks) throws MqttException
```

Comment:
Added a constructor in MqttConnection. However, as MqttConnection and Mqtt5Client is not in the same package, it is not possible to use a package-private config, instead I add a `getClientOptions` for Mqtt5Client so that the MqttConnection could access the config.

Reply:
Also, after thinking about it. This is a better workflow as it reduced the chance that the Mqtt5Client user to access the Mqtt3 interface. I will also update it for CPP and Python.